ISPN-15032 Use site and channel names as tag for metrics

pruivo opened the pull request:
https://issues.redhat.com/browse/ISPN-15032

When `names-as-tags` is set to `false`, we got 2 different metrics with the channel name distinguishing between them.

```
# TYPE vendor_jgroups_xsite_frag4_get_number_of_sent_fragments gauge
# HELP vendor_jgroups_xsite_frag4_get_number_of_sent_fragments Number of sent fragments
vendor_jgroups_xsite_frag4_get_number_of_sent_fragments{cluster="xsite",node="_pedro-laptop-4-17826:site1"} 0.0
# TYPE vendor_jgroups_cluster_frag4_get_number_of_sent_fragments gauge
# HELP vendor_jgroups_cluster_frag4_get_number_of_sent_fragments Number of sent fragments
vendor_jgroups_cluster_frag4_get_number_of_sent_fragments{cluster="cluster",node="pedro-laptop-4-17826"} 2.0
```

When `names-as-tags` is set to `true`, we got a single metric:

```
# TYPE vendor_jgroups_frag4_get_number_of_sent_fragments gauge
# HELP vendor_jgroups_frag4_get_number_of_sent_fragments Number of sent fragments
vendor_jgroups_frag4_get_number_of_sent_fragments{cache_manager="default",cluster="xsite",node="_pedro-laptop-4-16213:site1"} 0.0
vendor_jgroups_frag4_get_number_of_sent_fragments{cache_manager="default",cluster="cluster",node="pedro-laptop-4-16213"} 2.0
```

ryanemerson commented:
I see this is marked for 14.x, is it acceptable to change the behaviour of `name-as-tags` between patch releases?

pruivo commented:
@ryanemerson IMO, yes. Since the cluster name and the site name are dynamic it would make sense not to have those as metrics names. 

* Added a note to the upgrade guide. 
* Removed the label since it needs to be manually backported to 14.x

ryanemerson commented:
Thanks @pruivo
